dotnet/test/common/SeleniumManagerTest.cs

```diff
@@ -0,0 +1,97 @@
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+// </copyright>
+
+using NUnit.Framework;
+using OpenQA.Selenium.Chrome;
+using OpenQA.Selenium.Edge;
+using OpenQA.Selenium.Environment;
+using OpenQA.Selenium.Firefox;
+using OpenQA.Selenium.Safari;
+using System;
+using System.IO;
+
+namespace OpenQA.Selenium;
+
+[TestFixture]
+[Category("SeleniumManager")]
+[IgnoreBrowser(Browser.IE, "IE does not use Selenium Manager")]
+[IgnoreBrowser(Browser.Safari, "Safari does not need Selenium Manager")]
+[IgnoreBrowser(Browser.Remote, "Remote does not use Selenium Manager directly")]
+public class SeleniumManagerTest
+{
+    private static readonly string CacheDirectory = System.Environment.GetEnvironmentVariable("SE_CACHE") ?? Path.Combine(".cache", "selenium");
+
+    private DriverOptions CreateOptionsForCurrentBrowser()
+    {
+        return EnvironmentManager.Instance.Browser switch
+        {
+            Browser.Chrome => new ChromeOptions(),
+            Browser.Firefox => new FirefoxOptions(),
+            Browser.Edge => new EdgeOptions(),
+            _ => throw new NotSupportedException($"Browser {EnvironmentManager.Instance.Browser} is not supported for Selenium Manager tests")
+        };
+    }
+
+    private DriverService CreateServiceForCurrentBrowser(string driverPath)
+    {
+        return EnvironmentManager.Instance.Browser switch
+        {
+            Browser.Chrome => ChromeDriverService.CreateDefaultService(driverPath),
+            Browser.Firefox => FirefoxDriverService.CreateDefaultService(driverPath),
+            Browser.Edge => EdgeDriverService.CreateDefaultService(driverPath),
+            _ => throw new NotSupportedException($"Browser {EnvironmentManager.Instance.Browser} is not supported for Selenium Manager tests")
+        };
+    }
+
+    [Test]
+    public void ShouldGetDriverAndBrowserPaths()
+    {
+        var options = CreateOptionsForCurrentBrowser();
+        var driverFinder = new DriverFinder(options);
+
+        string driverPath = driverFinder.GetDriverPath();
+        string browserPath = driverFinder.GetBrowserPath();
+
+        Assert.That(File.Exists(driverPath), Is.True, $"Driver path should exist: {driverPath}");
+        Assert.That(File.Exists(browserPath), Is.True, $"Browser path should exist: {browserPath}");
+        Assert.That(driverPath, Does.Contain(CacheDirectory), $"Driver path should contain cache directory: {driverPath}");
+        Assert.That(browserPath, Does.Contain(CacheDirectory), $"Browser path should contain cache directory: {browserPath}");
```

Comment:
Seems like a product bug

```c#
Failed : OpenQA.Selenium.SeleniumManagerTest.ShouldGetDriverAndBrowserPaths
  Browser path should contain cache directory: C:\Program Files (x86)\Microsoft\Edge\Application\143.0.3650.96\msedge.exe
  Expected: String containing ".cache\selenium"
  But was:  "C:\Program Files (x86)\Microsoft\Edge\Application\143.0.3650.96\msedge.exe"
   at OpenQA.Selenium.SeleniumManagerTest.ShouldGetDriverAndBrowserPaths() in .\dotnet\test\common\SeleniumManagerTest.cs:line 74
```

Reply:
Well, the fact that it is looking for .cache\selenium instead of .cache/selenium needs to be fixed, but even with that...

I wonder if this means that .NET bazel doesn't accept --test_env values, I'll investigate.

Reply:
NUnit has the `Is.SubPathOf` constraint that should handle all the slashes for us.

Reply:
@RenderMichael can you put a code suggestion comment on this? https://github.com/SeleniumHQ/selenium/pull/16775/changes#diff-22f0b3f5b49032bf49ed44835bd02268e3eff2c82e3f06e7efd8b74a8df3d858R38

(or is it more complicated)

Reply:
The parent directory seems correct, we would just need to add a different assertion constraint. 

```suggestion
        Assert.That(browserPath, Is.SubPathOf(CacheDirectory), $"Browser path should be nested inside the cache directory: {browserPath}");
```

Reply:
Hmm, it seems to be pulling it properly in Mac at least.
@bonigarcia does SM print out in debug or trace which environment variables/config settings it is using?

@RenderMichael can you pull this branch locally and see if the values work for you on windows? (I need to create a Windows VM)

```
bazel test //dotnet/test/common:SeleniumManagerTest-chrome --test_env=SE_FORCE_BROWSER_DOWNLOAD=true --test_env=SE_SKIP_DRIVER_IN_PATH=true
```

Reply:
As a general rule, SM does not log the configuration values overridden with envs/config. But the debug/trace logs should be descriptive enough to follow what SM is doing.

Reply:
Would it make sense to log those, though? Perhaps someone added a config file and someone else can't figure out why it isn't behaving the way they think it should, etc?

Reply:
I believe the existing logs should be already descriptive enough. But if there is some specific configuration which is not clear, we can always include new logs

Reply:
I figured out what the issue was for this at least.